### PR TITLE
Focus Fixes for our Mobile Expression Keypad and MathInput 

### PR DESCRIPTION
**Pull request by SonicScrewdriver**
## Summary:
Ensured that the input span is focusable via tab navigation and that we cannot accidentally create a callback loop with our updated focus. 

I also implemented a couple of fixes to our full math input story so that it doesn't error when trying to callback, and blurs the input upon closing the keypad.

A separate fix will be implemented in WebApp that ensures that closing the keypad in an exercise moves the focus to the "Check Answer" button. It will be attached to the same ticket number (LC-1193) as it is directly related to the issue.

 ## Video:
https://github.com/Khan/perseus/assets/12463099/b5b418f6-64b6-41e5-9d6d-1a7c13410a29

Issue: LC-1193

## Test plan:
make tesc

### CODE REVIEW

File: packages/math-input/src/components/input/math-input.tsx [hunk 1]
```diff
@@ -908,11 +908,21 @@ class MathInput extends React.Component<Props, State> {
                 overrides.css. */}
                 <div
                     className="keypad-input"
-                    // @ts-expect-error - TS2322 - Type 'string' is not assignable to type 'number | undefined'.
-                    tabIndex={"0"}
+                    tabIndex={0}
```
Comment (SonicScrewdriver) [inline packages/math-input/src/components/input/math-input.tsx:911]: There was no need to have this ts error, and adding the index as a string was actually breaking our ability to focus on the span!

File: packages/math-input/src/components/input/math-input.tsx [hunk 2]
```diff
@@ -908,11 +908,21 @@ class MathInput extends React.Component<Props, State> {
...
                     ref={(node) => {
                         this.inputRef = node;
                     }}
+                    onFocus={() => {
+                        this.focus();
+                    }}
+                    onBlur={() => {
```
Comment (SonicScrewdriver) [inline packages/math-input/src/components/input/math-input.tsx:918]: I will be making a separate ticket and PR that adds logic to ensure that the mathInput retains the focus styles when any element within the keypad is the activeElement. Unfortunately Storybook really breaks being able to test "activeElement" due to the use of some strange iframes, so we've decided to tackle this issue post launch.

Comment (SonicScrewdriver) [reply packages/math-input/src/components/input/math-input.tsx:918]: https://khanacademy.atlassian.net/browse/LC-1209

File: packages/math-input/src/components/input/math-input.tsx [hunk 3]
```diff
@@ -908,11 +908,21 @@ class MathInput extends React.Component<Props, State> {
...
+                        this._hideCursorHandle();
+                        this.mathField.blur();
+                        this.setState({
+                            focused: false,
+                            handle: {visible: false},
+                        });
```
Comment (handeyeco) [inline packages/math-input/src/components/input/math-input.tsx:924]: I think we'll just want to call `this.blur()` here right?
```suggestion
                        this.blur();
```

File: packages/math-input/src/full-math-input.stories.tsx [hunk 4]
```diff
@@ -73,10 +75,13 @@ export const Basic = () => {
...
             <KeypadInput
                 value={value}
+                ref={input}
                 keypadElement={keypadElement}
                 onChange={(newValue, callback) => {
                     setValue(newValue);
-                    callback();
+                    if (callback) {
+                        callback();
+                    }
```
Comment (handeyeco) [inline packages/math-input/src/full-math-input.stories.tsx:84]: Just a nit
```suggestion
                        callback?.();
```

**Review by handeyeco — APPROVED** [review]
I'd like it if we could reuse the `blur` method in `math-input`, but this otherwise looks good to go!

File: packages/math-input/src/components/input/math-input.tsx (hunk 3 above)
Comment (SonicScrewdriver) [reply packages/math-input/src/components/input/math-input.tsx:924]: Calling this.blur actually causes us to call our parent methods, which causes the keypad to close undesirably:

` this.props.onBlur && this.props.onBlur();`

However I could make this.blur() take a boolean that conditionally runs this line if that's preferrable :)

**Pull request merged**